### PR TITLE
Workaround native image charset and kotlin issue

### PR DESCRIPTION
**Pull request by yschimke**
```
      ├─ noRecoveryFromTwoRefusedStreams(Protocol, MockWebServer) ✔
      │  ├─ [1] h2_prior_knowledge ✘ Could not initialize class kotlin.internal.PlatformImplementationsKt
      │  └─ [2] h2 ✘ Could not initialize class kotlin.internal.PlatformImplementationsKt
```

and

```
   │  ├─ readerBomUtf16Be() ✔
   │  ├─ readerBomUtf16Le() ✔
   │  ├─ readerBomUtf32Be() ✘ UTF-32BE
   │  ├─ readerBomUtf32Le() ✘ UTF-32LE
```

### CODE REVIEW

**Comment by yschimke** [comment]
With this


```
   ├─ RouteExceptionTest ✔
   │  ├─ getConnectionIOException_single() ✔
   │  └─ getConnectionIOException_multiple() ✘
   │           Expecting:
   │            <0>
   │           to be equal to:
   │            <2>
   │           but was not.
   └─ HttpOverHttp2Test ✔
      ├─ concurrentHttp2ConnectionsDeduplicated(Protocol, MockWebServer) ✔
      │  ├─ [1] h2_prior_knowledge ✘ got: <false>, expected: is <true>
      │  └─ [2] h2 ✔
      ├─ missingPongsFailsConnection(Protocol, MockWebServer) ✔
      │  ├─ [1] h2_prior_knowledge ✔
      │  └─ [2] h2 ✘ Cannot determine correct type for matchesSafely() method.

Test run finished after 46943 ms
[        57 containers found      ]
[         1 containers skipped    ]
[        56 containers started    ]
[         0 containers aborted    ]
[        56 containers successful ]
[         0 containers failed     ]
[       108 tests found           ]
[         0 tests skipped         ]
[       108 tests started         ]
[         0 tests aborted         ]
[       105 tests successful      ]
[         3 tests failed          ]
```

File: native-image-tests/build.gradle [hunk 1]
```diff
@@ -69,6 +69,7 @@ graal {
   option "-H:+TraceClassInitialization"
   option "--initialize-at-build-time=kotlin.text.Charsets"
   option "-H:EnableURLProtocols=http,https"
+  option "-H:+AddAllCharsets"
```
Comment (swankjesse) [inline native-image-tests/build.gradle:72]: We should document which of these options our users need for OkHttp to behave correctly. For example, I suspect `TraceClassInitialization` is only for testing, but `kotlin.text.Charsets` is necessary for production? (and if it is, should it be?)

Comment (swankjesse) [reply native-image-tests/build.gradle:72]: Is there a way for us to promote `--initialize-at-build-time=kotlin.text.Charsets` into one of our JSON files? The more stuff that’s in those, the more stuff we can change in the future without breaking consumers.

**Review by swankjesse — APPROVED** [review]

File: native-image-tests/build.gradle (hunk 1 above)
Comment (yschimke) [reply native-image-tests/build.gradle:72]: I think kotlin.text.Charsets is a bug, probably came out of work I'm doing in the AutomaticFeature.

**Pull request merged**